### DISCUSSION TITLE
GetFiles - GlobPattern for exclusion?

### DISCUSSION THREAD

**Discussion by dd-eg** [cake-build/cake#4118] [Q&A]
I have a directory `Folder` containing files like this:
```
File.dll
File.pdb
File.config
```
I can use `GetFiles("Folder//*(dll|config)")` if I want to call out individual extensions to retrieve. However, what I REALLY want is to specifically exclude .pdb files. I am looking for something like `GetFiles("Folder//*!(pdb)")`, but every variation I have tried does not work. I know there is an overload that takes `GlobeSettings` where a predicate can be used, but this will not work for my purposes (my input is only a single string).

What is the expected syntax for this (or is it not supported)?

**Comment by augustoproiete** [comment]
@dd-eg The syntax to get all files except for `.pdb` would be `GetFiles("Folder//*.*[ˆpdb]") `

**Reply by dd-eg** [reply]
When I test `GetFiles("Folder//*.*[ˆpdb]")`, I actually see the opposite behavior, i.e. I get only `.pdb` files.

If I test `GetFiles("Folder//*.*[!pdb]")`, then I do only get `.pdb` files for this particular example, but this appears to be acting similar to a character range match instead of a string match -- if there are `.mdb` files present, then those will be excluded as well.

**Reply by patriksvensson** [reply]
Afaik, this is not possible. It been a while since I wrote the globber but I think this is an unsupported scenario. 

You will need to do something like `var result = GetFiles(”foo”) - GetFiles(”bar”)`

**Reply by dd-eg** [reply]
Ah well. Thanks for the responses everyone.

Is this worth me raising an issue (improvement)?
